shared/origami/CMakeLists.txt

```
@@ -0,0 +1,156 @@
+# Copyright Advanced Micro Devices, Inc., or its affiliates.
+# SPDX-License-Identifier:  MIT
+
+cmake_minimum_required(VERSION 3.24.4)
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    project(Origami VERSION 1.0.0 LANGUAGES CXX)
+
+    set(ORIGAMI_STANDALONE ON)
+else()
+    set(ORIGAMI_STANDALONE OFF)
+endif()
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(GenerateExportHeader)
+include(CMakeDependentOption)
+
+include(dependencies)
+
+rocm_setup_version(VERSION ${PROJECT_VERSION})
+
+option(ORIGAMI_BUILD_SHARED_LIBS "Build shared libraries." ON)
+option(ORIGAMI_ENABLE_PYTHON "Enable Python bindings." OFF)
+option(ORIGAMI_BUILD_TESTING "Enable Python binding tests." OFF)
+
+find_package(hip REQUIRED)
+
+if(ORIGAMI_BUILD_SHARED_LIBS)
+    add_library(origami SHARED) 
+else()
+    add_library(origami STATIC)
+endif()
+
+rocm_set_soversion(origami ${PROJECT_VERSION_MAJOR})
+
+add_library(origami::origami ALIAS origami)
+
+set_target_properties(origami PROPERTIES
+    POSITION_INDEPENDENT_CODE ON
+)
+
+target_compile_features(origami PUBLIC cxx_std_17)
+
+target_include_directories(origami
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+
+target_sources(origami
+    PRIVATE
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/gemm.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/hardware.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/utils.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/origami/streamk.cpp"
+)
+
+target_link_libraries(origami PUBLIC hip::host)
+
+if(ORIGAMI_ENABLE_PYTHON)
+    find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+    find_package(Python 3.8 COMPONENTS Interpreter Development.Module REQUIRED)
+
+    include(FetchContent)
+    FetchContent_Declare(
+        nanobind
+        GIT_REPOSITORY https://github.com/wjakob/nanobind.git
+        GIT_TAG        v1.9.2  # or latest stable version
+    )
+    FetchContent_MakeAvailable(nanobind)
+    find_package(nanobind CONFIG REQUIRED)
+
+    nanobind_add_module(origami_python 
+        "${CMAKE_CURRENT_SOURCE_DIR}/python/origami_module.cpp"
+    )
+    
+    set_target_properties(origami_python PROPERTIES
+        OUTPUT_NAME "origami"
+    )
+    
+    target_link_libraries(origami_python PRIVATE origami::origami)
+endif()
```

A thought
```suggestion
if(ORIGAMI_ENABLE_PYTHON)
     add_subdirectory(Python)
 endif()
```

And move this logic into a CMakeLists.txt in the Python directory.